File: swift/tests/BedrockTests/BedrockTestSupport.swift
import Foundation
@testable import Bedrock

/// Shared Bedrock initialization for the test suites.
///
/// `setConfig` is a one-shot global that refuses every call after the first, so this
/// helper is what makes it safe to call from each suite's `setUp`.
enum BedrockTestSupport {
    static let rootPath: String = {
        let root = FileManager.default.temporaryDirectory
            .appendingPathComponent("bedrock-tests-\(ProcessInfo.processInfo.processIdentifier)")
        // clear anything left from a prior run.
        try? FileManager.default.removeItem(at: root)
        return root.path
    }()

    /// Initializes the global Bedrock config, if it isn't already.
    ///
    /// A second `setConfig` throws, and this runs before every test, so the guard is what
    /// keeps every test after the first from failing in `setUp`.
    static func setUp() throws {
        guard !isInitialized() else { return }
        try Bedrock.setConfig(environment: .staging, os: .ios, rootPath: rootPath)
    }
}

File: kotlin/bedrock-tests/src/test/kotlin/bedrock/BedrockTestSupport.kt
package bedrock

import java.nio.file.Files

/**
 * Shared Bedrock initialization for the test suites.
 *
 * `setConfig` is a one-shot global that refuses every call after the first, so this
 * helper is what makes it safe to call from each suite's `setUp`.
 */
object BedrockTestSupport {
    /** Root directory Bedrock writes to during the test run. Recreated per process. */
    val dataDirectory: String by lazy {
        val root = Files.createTempDirectory("bedrock-tests").toFile()
        Runtime.getRuntime().addShutdownHook(Thread { root.deleteRecursively() })
        root.absolutePath
    }

    /**
     * Initializes the global Bedrock config, if it isn't already.
     *
     * A second `setConfig` throws, and this runs before every test, so the guard is what
     * keeps every test after the first from failing in `setUp`.
     */
    fun setUp() {
        if (uniffi.bedrock.isInitialized()) return

        uniffi.bedrock.setConfig(
            uniffi.bedrock.BedrockEnvironment.STAGING,
            uniffi.bedrock.Os.ANDROID,
            dataDirectory,
        )
    }
}
